### PR TITLE
Support --default-repo=‘’ to erase the value from global config

### LINKED ISSUE #3923: unable to set an empty default repo
I have an issue during `skaffold build` that I cannot specify an empty value "" as the `default-repo`.
The parameter `--default-repo=""` doesn't seem to be taking effect.

In my skaffold yaml I have two profiles, one with the "base images" and the other with the application services.
The base images are derived from CentOS and then updated and needed packages are added to them.
All the application service use the base image created as part of the base images.

### Steps
1. skaffold build -p base_images -b mycentos
2. skaffold build -p application -b myservice1

### global skaffold config
`cat ~/.skaffold/config`
```
  default-repo: my-private-ip:5000/shantanu
```

I tried the following cmd, but that doesn't work either:
` env SKAFFOLD_DEFAULT_REPO="" skaffold build -p base_images -b mycentos --default-repo ""`

### Expected behavior
A Docker image with just the name `mycentos:latest` should be created.
My application service images are using the custom `mycentos:latest` in their `FROM` line

### Actual behavior
A docker image get created with the full tag gets created:
```
docker images

my-private-ip:5000/shantanu/mycentos   latest ...

```

### Information

- Skaffold version: v1.7.0
- Operating system: Ubuntu 16.04.x

### PR DESCRIPTION
**Pull request by dgageot**
Fixes #3923

Signed-off-by: David Gageot <david@gageot.net>

### CODE REVIEW

File: cmd/skaffold/app/cmd/cmd.go [hunk 1]
```diff
@@ -81,6 +81,12 @@ func NewSkaffoldCommand(out, err io.Writer) *cobra.Command {
 				opts.Force = true
 			}
 
+			// Allows empty default-repo flag to passed on the command-line
```
Comment (nkubala) [inline cmd/skaffold/app/cmd/cmd.go:84]:
```suggestion
			// Allows empty default-repo flag to be passed on the command-line
```

File: pkg/skaffold/config/options.go [hunk 2]
```diff
@@ -35,6 +35,8 @@ type PortForwardOptions struct {
 type SkaffoldOptions struct {
 	ConfigurationFile  string
 	GlobalConfig       string
+	DefaultRepo        string
```
Comment (nkubala) [inline pkg/skaffold/config/options.go:38]: is it possible to make this a `*string` so we can differentiate between `nil` and `""`? then we wouldn't need the `DefaultRepoSet` boolean

Comment (dgageot) [reply pkg/skaffold/config/options.go:38]: That's what I've tried first but couldn't make it work. I'll try again

Comment (dgageot) [reply pkg/skaffold/config/options.go:38]: nope, couldn't find a way to do it

File: pkg/skaffold/config/options.go [hunk 3]
```diff
@@ -35,6 +35,8 @@ type PortForwardOptions struct {
...
+	DefaultRepoSet     bool
```
Comment (briandealwis) [inline pkg/skaffold/config/options.go:39]: `Set` is awkward. `Overridden`?  Or would it be easier to use a magic value (`<<unset>>`).

File: pkg/skaffold/config/util.go [hunk 4]
```diff
@@ -147,9 +147,9 @@ func getConfigForKubeContextWithGlobalDefaults(cfg *GlobalConfig, kubeContext st
 	return &mergedConfig, nil
 }
 
-func GetDefaultRepo(configFile, cliValue string) (string, error) {
+func GetDefaultRepo(configFile, cliValue string, cliValueSet bool) (string, error) {
 	// CLI flag takes precedence.
-	if cliValue != "" {
+	if cliValueSet {
```
Comment (briandealwis) [inline pkg/skaffold/config/util.go:152]: ditto `Set` -> `Override`?

File: cmd/skaffold/app/cmd/cmd.go [hunk 5]
```diff
@@ -81,6 +81,12 @@ func NewSkaffoldCommand(out, err io.Writer) *cobra.Command {
 				opts.Force = true
 			}
 
+			// Allows empty default-repo flag to be passed on the command-line
```
Comment (briandealwis) [inline cmd/skaffold/app/cmd/cmd.go:84]: `// to revert to values in skaffold.yaml`

**Comment by dgageot** [comment]
@briandealwis @nkubala I came up with a nicer solution, I think. Feels much less hacky

File: pkg/skaffold/config/util.go [hunk 6]
```diff
@@ -147,10 +147,10 @@ func getConfigForKubeContextWithGlobalDefaults(cfg *GlobalConfig, kubeContext st
 	return &mergedConfig, nil
 }
 
-func GetDefaultRepo(configFile, cliValue string) (string, error) {
+func GetDefaultRepo(configFile string, cliValue *string) (string, error) {
 	// CLI flag takes precedence.
-	if cliValue != "" {
-		return cliValue, nil
+	if cliValue != nil {
```
Comment (briandealwis) [inline pkg/skaffold/config/util.go:152]: We should have a test specifically for GetDefaultRepo.

**Comment by dgageot** [comment]
Let me merge that and add a few more tests. Maybe also change `GetLocalCluster` to behave the same

**Pull request merged**